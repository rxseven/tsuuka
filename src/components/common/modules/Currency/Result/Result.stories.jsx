import React from 'react';
import { storiesOf } from '@storybook/react';

import Result from './index';

const title = 'Components/Common/Modules/Currency/Result';
const props = {
  baseCurrency: { code: 'USD', name: 'United States Dollar' },
  countries: [{ code: 'TH', name: 'Thailand' }],
  rate: '31.7916',
  targetCurrency: { code: 'THB', name: 'Thai Baht' },
  timestamp: 1551581826
};

storiesOf(title, module)
  .add('one item', () => <Result {...props} />)
  .add('multiple items', () => (
    <Result
      {...props}
      countries={[
        { code: 'DE', name: 'Germany' },
        { code: 'ES', name: 'Spain' },
        { code: 'IT', name: 'Italy' }
      ]}
      rate="0.87892"
      targetCurrency={{ code: 'EUR', name: 'Euro' }}
    />
  ));
