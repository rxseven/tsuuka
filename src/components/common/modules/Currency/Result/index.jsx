import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import moment from 'moment';
import React from 'react';
import styled from 'styled-components';

import Hyperlink from 'components/common/base/Hyperlink';
import Country from 'components/common/sections/Country';

const Highlight = styled.i`
  color: ${({ theme }) => theme.color.primary};
  font-style: normal;
`;

const Meta = styled.div`
  border-top: #beabda solid 1px;
  color: ${({ theme }) => theme.color.secondary};
  font-size: 0.75rem;
  margin-top: 1rem;
  padding-top: 1rem;
`;

const propTypes = exact({
  baseCurrency: PropTypes.shape({
    code: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired,
  countries: PropTypes.arrayOf(
    PropTypes.shape({
      code: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    })
  ).isRequired,
  rate: PropTypes.string.isRequired,
  targetCurrency: PropTypes.shape({
    code: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired,
  timestamp: PropTypes.number.isRequired
});

function Result(props) {
  const { baseCurrency, countries, rate, targetCurrency, timestamp } = props;
  const countriesLength = countries.length;

  return (
    <div data-testid="result">
      <p>
        <Highlight>1 {baseCurrency.code}</Highlight> is worth{' '}
        <Highlight>
          {rate} {targetCurrency.code}
        </Highlight>
        . You can spend it{' '}
        {countriesLength > 1
          ? `in the
    following ${countriesLength} countries`
          : 'only in'}
        :
      </p>
      <Country data={countries} />
      <Meta>
        Last updated{' '}
        <Hyperlink href="https://fixer.io" title="Fixer.io">
          foreign exchange rates
        </Hyperlink>{' '}
        at{' '}
        <strong data-testid="latest-date">
          {moment.unix(timestamp).format('HH:mm:ss')}
        </strong>{' '}
        on{' '}
        <strong data-testid="latest-time">
          {moment.unix(timestamp).format('DD.MM.YYYY')}
        </strong>{' '}
        (free plan,{' '}
        <Hyperlink
          href="https://fixer.io/product"
          title="Fixer's Plans & Pricing"
        >
          hourly updates
        </Hyperlink>
        ).
      </Meta>
    </div>
  );
}

Result.propTypes = propTypes;

export default Result;
