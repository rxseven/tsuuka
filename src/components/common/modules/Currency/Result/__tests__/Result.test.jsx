import { shallow } from 'enzyme';
import React from 'react';

import Environment from 'tests/environment';
import Result from '../index';

describe('<Currency.Result />', () => {
  // Arrange
  const props = {
    baseCurrency: { code: 'USD', name: 'United States Dollar' },
    countries: [{ code: 'TH', name: 'Thailand' }],
    rate: '31.7916',
    targetCurrency: { code: 'THB', name: 'Thai Baht' },
    timestamp: 1551581826
  };
  const component = (
    <Environment>
      <Result {...props} />
    </Environment>
  );

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });
});
