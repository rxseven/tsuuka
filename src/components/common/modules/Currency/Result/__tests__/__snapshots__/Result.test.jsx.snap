// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Currency.Result /> Snapshot tests should render correctly 1`] = `
Array [
  <p>
    <i
      className="sc-ifAKCX jEgmKQ"
    >
      1 
      USD
    </i>
     is worth
     
    <i
      className="sc-ifAKCX jEgmKQ"
    >
      31.7916
       
      THB
    </i>
    . You can spend it
     
    only in
    :
  </p>,
  <ul
    className="sc-htpNat eZEtUy"
  >
    <li
      className="sc-bwzfXH hQcoUX"
    >
      <img
        alt="Thailand"
        className="sc-bdVaJa gChJQj"
        src="https://www.countryflags.io/TH/flat/16.png"
      />
      <span
        className="sc-bxivhb hMCZnY"
      >
        Thailand
      </span>
    </li>
  </ul>,
  <div
    className="sc-EHOje fdDCzn"
  >
    Last updated
     
    <a
      href="https://fixer.io"
      rel="noopener noreferrer"
      target="_blank"
      title="Fixer.io"
    >
      foreign exchange rates
    </a>
     
    at 
    <strong>
      09:57:06
    </strong>
     on
     
    <strong>
      03.03.2019
    </strong>
     (free plan,
     
    <a
      href="https://fixer.io/product"
      rel="noopener noreferrer"
      target="_blank"
      title="Fixer's Plans & Pricing"
    >
      hourly updates
    </a>
    ).
  </div>,
]
`;
